library/pool.c

```diff
@@ -276,10 +278,31 @@ static inline void make_thread_stat_id(char *buffer, int size, struct dnet_work_
 	snprintf(buffer, size - 1, "%s.%s", pool->pool_id, mode_marker);
 }
 
+struct dnet_cmd *dnet_io_req_get_cmd(struct dnet_io_req *r) {
+	if (r->io_req_type == DNET_IO_REQ_OLD_PROTOCOL)
+		// dnet_io_req enqueued from old mechanic
+		return r->header;
+	else
+		// dnet_io_req enqueued from protocol-independent mechanic
+		return n2_io_req_get_cmd(r);
+}
+
+int dnet_io_req_set_request_backend_id(struct dnet_io_req *r, int backend_id) {
```

Comment:
Why does it return an error code when caller ignores it?

Reply:
Also how about moving this function into `protocol.cpp` and merging with `n2_io_req_set_request_backend_id`?

Reply:
> Why does it return an error code when caller ignores it?

Maybe I should process that error code in caller and at least to log it. Anyway, function _must_ have a way to report about error. In some cases, for example in destructor, errors of subroutines sometimes ignored, but subroutines must report - not their deal do decide for high-level logic.

Reply:
> Also how about moving this function into protocol.cpp

I didn't plan to upgrade `protocol.cpp` anyway. `protocol.cpp` will wane, and `n2_protocol` will grow instead. Finally we can rename `n2_protocol` to `protocol`, deleting the previous one. I think it's better not to mix new logic into old, waning code.

`dnet_io_req_set_request_backend_id` is temporary function, it supports legacy. `n2_io_req_set_request_backend_id` will stay at the end.